keyboards/1upkeyboards/pi50/pi50.c

```diff
@@ -0,0 +1,290 @@
+            break;
+        case 21:
+            oled_write_P(PSTR("Rainbow Beacon\n                  "), false);
+            break;
+        case 22:
+            oled_write_P(PSTR("Rainbow Pinwheels\n                  "), false);
+            break;
+        case 23:
+            oled_write_P(PSTR("Raindrops\n                  "), false);
+            break;
+        case 24:
+            oled_write_P(PSTR("Jellybean Raindrops\n                  "), false);
+            break;
+        case 25:
+            oled_write_P(PSTR("Hue Breathing\n                  "), false);
+            break;
+        case 26:
+            oled_write_P(PSTR("Hue Pendulum\n                  "), false);
+            break;
+        case 27:
+            oled_write_P(PSTR("Hue Wave\n                  "), false);
+            break;
+        case 28:
+            oled_write_P(PSTR("Pixel Rain\n                  "), false);
+            break;
+        case 29:
+            oled_write_P(PSTR("Pixel Flow\n                  "), false);
+            break;
+        case 30:
+            oled_write_P(PSTR("Pixel Fractal\n                  "), false);
+            break;
+        case 31:
+            oled_write_P(PSTR("Typing Heatmap\n                  "), false);
+            break;
+        case 32:
+            oled_write_P(PSTR("Digital Rain\n                  "), false);
+            break;
+        case 33:
+            oled_write_P(PSTR("Solid Reactive\nSimple            "), false);
+            break;
+        case 34:
+            oled_write_P(PSTR("Solid Reactive\n                  "), false);
+            break;
+        case 35:
+            oled_write_P(PSTR("Solid Reactive\nWide              "), false);
+            break;
+        case 36:
+            oled_write_P(PSTR("Solid Reactive\nMultiwide         "), false);
+            break;
+        case 37:
+            oled_write_P(PSTR("Solid Reactive\nCross             "), false);
+            break;
+        case 38:
+            oled_write_P(PSTR("Solid Reactive\nMulticross        "), false);
+            break;
+        case 39:
+            oled_write_P(PSTR("Solid Reactive\nNexus             "), false);
+            break;
+        case 40:
+            oled_write_P(PSTR("Solid Reactive\nMultinexus        "), false);
+            break;
+        case 41:
+            oled_write_P(PSTR("Splash\n                  "), false);
+            break;
+        case 42:
+            oled_write_P(PSTR("Multisplash\n                  "), false);
+            break;
+        case 43:
+            oled_write_P(PSTR("Solid Splash\n                  "), false);
+            break;
+        case 44:
+            oled_write_P(PSTR("Solid Multisplash\n                  "), false);
+            break;
+        default:
+            // Or use the write_ln shortcut over adding '\n' to the end of your string
+            oled_write_ln_P(PSTR("Undefined\n                  "), false);
+    }
```

Comment:
There is a better way to handle this, since these numbers may actually change, if a users enables/disables modes or enables custom kb/user animations. 

This code is respectfully stolen from tzarc, and will generate the name based off the function names.  And it will automatically account for any enabled or disabled effects.

https://github.com/drashna/qmk_firmware/blob/480865713eeeb1a0fa38e49de02d49370e1bf580/users/drashna/painter/ili9341_display.c#L37-L72

you'd place that before this function, and then you can call it with:
```c
    oled_write_ln(rgb_matrix_name(rgb_matrix_get_mode()), false;
```

Reply:
I added the requested changes